lighthouse-core/gather/gatherers/trace-nodes.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license Copyright 2020 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+const pageFunctions = require('../../lib/page-functions.js');
+const TraceProcessor = require('../../lib/tracehouse/trace-processor.js');
+
+/**
+ * @return {LH.Artifacts['TraceNodes']}
+ */
+/* istanbul ignore next */
+function collectTraceNodes() {
+  /** @type {Array<HTMLElement>} */
+  // @ts-ignore - put into scope via stringification
+  const markedElements = getElementsInDocument('[lhtemp]'); // eslint-disable-line no-undef
+  /** @type {LH.Artifacts['TraceNodes']} */
+  const traceNodes = [];
+
+  for (const element of markedElements) {
+    // @ts-ignore - put into scope via stringification
+    const htmlSnippet = getOuterHTMLSnippet(element); // eslint-disable-line no-undef
+    traceNodes.push({
+      metricTag: element.getAttribute('lhtemp') || '',
+      // @ts-ignore - put into scope via stringification
+      nodePath: getNodePath(element), // eslint-disable-line no-undef
+      // @ts-ignore - put into scope via stringification
+      selector: getNodeSelector(element), // eslint-disable-line no-undef
+      // @ts-ignore - put into scope via stringification
+      nodeLabel: getNodeLabel(element), // eslint-disable-line no-undef
+      snippet: htmlSnippet.replace(' lhtemp="lcp"', ''),
```

Comment:
Let's be good custodians and clean up the changes we made to the DOM. then you can also avoid this replace.

Reply:
Took out the logic for removing the temp attribute when I decoupled from the CLS work. However, this replace statement is still necessary as even after removing the attribute, the snippet will still contain it since it was there at the time we grabbed the element.

Reply:
you could remove the attribute before `const htmlSnippet = ...`